src/command_modules/azure-cli-eventgrid/azure/cli/command_modules/eventgrid/tests/test_eventgrid_commands.py

```diff
@@ -0,0 +1,128 @@
+
+    @ResourceGroupPreparer()
+    @StorageAccountPreparer()
+    def test_create_event_subscriptions_to_resource(self, resource_group, storage_account):
+        event_subscription_name = self.create_random_name(prefix='cli', length=40)
+
+        self.cmd('az eventgrid resource event-subscription create -g {} --provider-namespace Microsoft.Storage --resource-type storageAccounts --resource-name {} --name {} --endpoint https://requestb.in/18zmdhv1'.format(resource_group, storage_account, event_subscription_name), checks=[
+            JMESPathCheck('type', 'Microsoft.EventGrid/eventSubscriptions'),
+            JMESPathCheck('provisioningState', 'Succeeded'),
+            JMESPathCheck('name', event_subscription_name),
+        ])
+        self.cmd('az eventgrid resource event-subscription show -g {} --provider-namespace Microsoft.Storage --resource-type storageAccounts --resource-name {} --name {}'.format(resource_group, storage_account, event_subscription_name), checks=[
+            JMESPathCheck('type', 'Microsoft.EventGrid/eventSubscriptions'),
+            JMESPathCheck('provisioningState', 'Succeeded'),
+            JMESPathCheck('name', event_subscription_name),
+        ])
+        self.cmd('az eventgrid resource event-subscription show-endpoint-url -g {} --provider-namespace Microsoft.Storage --resource-type storageAccounts --resource-name {} --name {}'.format(resource_group, storage_account, event_subscription_name), checks=[
+            JMESPathCheck('endpointUrl', 'https://requestb.in/18zmdhv1'),
+        ])
+        self.cmd('az eventgrid resource event-subscription list -g {} --provider-namespace Microsoft.Storage --resource-type storageAccounts --resource-name {}'.format(resource_group, storage_account), checks=[
+            JMESPathCheck('[0].type', 'Microsoft.EventGrid/eventSubscriptions'),
+            JMESPathCheck('[0].provisioningState', 'Succeeded'),
+        ])
+        self.cmd('az eventgrid resource event-subscription delete -g {} --name {} --provider-namespace Microsoft.Storage --resource-type storageAccounts --resource-name {}'.format(resource_group, event_subscription_name, storage_account))
+
+    @ResourceGroupPreparer()
+    def test_create_event_subscriptions_with_filters(self, resource_group):
+        event_subscription_name = 'eventsubscription2'
+        ENDPOINT_URL = 'https://requestb.in/18zmdhv1'
+        SUBJECT_ENDS_WITH = 'mysubject_suffix'
+        EVENT_TYPE_1 = 'blobCreated'
+        EVENT_TYPE_2 = 'blobUpdated'
+        LABEL_1 = 'Finance'
+        LABEL_2 = 'HR'
+
+        self.cmd('az eventgrid event-subscription create -g {} --name {} --endpoint {} --subject-ends-with {} --included-event-types {} {} --is-subject-case-sensitive --labels {} {}'.format(resource_group, event_subscription_name, ENDPOINT_URL, SUBJECT_ENDS_WITH, EVENT_TYPE_1, EVENT_TYPE_2, LABEL_1, LABEL_2))
```

Comment:
Recommend aliasing `--is-subject-case-sensitive` to `--subject-case-sensitive` or `--subject-is-case-sensitive` (I prefer the shorter one!).  The reason is the first form doesn't sound natural as a flag--it seems to suggest there must be a true or false afterwards. The two alternatives are self-explanatory as flags or (obviously) with true/false.